alacritty/src/event.rs

```diff
@@ -844,10 +844,10 @@ impl<'a, N: Notify + 'a, T: EventListener> ActionContext<'a, N, T> {
         // Clear focused match.
         self.search_state.focused_match = None;
 
-        // The viewport reset logic is only needed for vi mode, since without it our origin is
-        // always at the current display offset instead of at the vi cursor position which we need
-        // to recover to.
-        if !self.terminal.mode().contains(TermMode::VI) {
+        // The viewport reset logic is only needed for vi mode with an active search, since without
+        // it the terminal has no dirty search contents and our origin is always at the current
+        // display offset instead of at the vi cursor position which we need to recover to.
+        if !(self.terminal.mode().contains(TermMode::VI) && self.search_active()) {
```

Comment:
Could you simplify this logical expression? I think clippy should suggest using `or` instead of `and` here?